packages/connect/src/Detector.ts

```diff
@@ -100,6 +100,24 @@ export class Detector {
    * {@link https://polkadot.js.org/docs/}
    */
   public connect = async (chainName: string, chainSpec?: string, options?: ApiOptions): Promise<ApiPromise> => {
+    const provider: ExtensionProvider | SmoldotProvider = this.provider(chainName, chainSpec);
+    provider.connect().catch(console.error);
+
+    this.#providers[chainName] = provider as ProviderInterface;
+    return await ApiPromise.create(Object.assign(options ?? {}, {provider}));
+  }
+
+  /**
+   * creates and returns a provider (either Smoldot or Extension one) to be used
```

Comment:
"Detects and returns an appropriate PolkadotJS provider depending on whether the user has the substrate connect extension installed"